### ISSUE TITLE
Allow translation for Website Inbox Welcome Heading, Welcome Tagline and Channel greeting message

### ISSUE THREAD

**Issue by sebastienlabine** [chatwoot/chatwoot#1374]
**Is your feature request related to a problem? Please describe.**
We have users who speak both french and english.  Would love to have the website popup chat greet them in their respective lang based on HTML Lang attribute or Url path (/fr/..)

**Describe the solution you'd like**
We would like to have a translation tab for each language we want to support and be able to edit each key for this particular language in the inbox setting tab.

**Describe alternatives you've considered**
Only greet in a single language.

**Comment by pranavrajs** [comment]
@sebastienlabine We have thought about this. One of the things we are considering to do is to move the exisiting settings to SDK, so that along with locale you can pass the welcome title and tagline. This would be done in future releases.

**Comment by pklreak** [comment]
Also, maybe add the collect email input text in the SDK, to make sure all text would be configurable

**Comment by apvlv** [comment]
Hello, 

is this still planned?

Thank you very much.

**Comment by apvlv** [comment]
bump

**Comment by gtmadev** [comment]
Options via the JS SDK would be ideal. That way, you can implement any client-side i18n solution and set the options via those frameworks. Would yield the greatest flexibility.

**Comment by matbgn** [comment]
Any news for this since 2020? We are in a country that has 4 national languages, customer support will be fairy tail for us if you implement this ;-)

**Comment by nickkuijpers** [comment]
Would love this also. To be able to change the text when the e-mail is requested in the livechat.

**Comment by jaime-rivas** [comment]
Agree, this is an important feature for multilingual service. @pranavrajs any updates on this? Thank you so much!

**Comment by hheexx** [comment]
We also need this to implement chatwoot. 
+1

**Comment by srflorea1** [comment]
We need it too

**Comment by chenliu9** [comment]
Hey @pranavrajs my team recently started using Chatwoot, and we're genuinely impressed with its capabilities and user-friendliness. We'd love to having this feature, and it will significantly enhance our experience with Chatwoot.

**Comment by pmrotule** [comment]
It's sad to see that this feature has still not been prioritized after 3 years. Chatwoot is such a great tool 🙌 , but we're thinking of switching to another provider just because we can't define the language of the welcome message and other automated messages like the one triggered outside of business hours. This is a must-have in my opinion.

**Comment by pmrotule** [comment]
I just saw Chatwoot is written in Rails and Vue which I'm familiar with. I might be willing to contribute. Looking at the current implementation, that's what I would have in mind:

_I'm talking specifically about `out_of_office_message`, but a similar approach could be used for the welcome message_

- Instead of using the `out_of_office_message` field in the `Inbox` model, we create a new translated field called `translated_out_of_office_message` using the [`globalize`](https://github.com/globalize/globalize) or [`mobility`](https://github.com/shioyama/mobility) gem
- In the frontend, we can render a `<woot-message-editor>` for each available language together with a heading specifying which language you're editing

https://github.com/chatwoot/chatwoot/blob/e3eca47c31b981721b3280df6c0d9db6b7b94b51/app/javascript/dashboard/routes/dashboard/settings/inbox/components/WeeklyAvailability.vue#L29-L30

- When sending the value of `unavailableMessage` with the POST request to `updateInbox`, we send a JSON string with the language as key and message as value, and use the `translated_out_of_office_message` field instead
- In the backend within `Api::V1::Accounts::InboxesController#update`, we `JSON.parse` the value of `translated_out_of_office_message` and store it the different messages in the database depending on the translation gem we chose
- Figure out a way to pass the current locale to `MessageTemplates::Template::OutOfOffice` so that we render the right message

Would that make sense?

**Comment by darker0n** [comment]
Hello everyone! I faced the same problem as you and found a solution. You can create a separate inbox for each required language with a translated greeting message, heading, and tagline. And connect the same agents to each inbox. Then just add conditional in javascript code on your website like this:
```
<script>
const path = window.location.pathname;
let websiteToken = 'default_token'; // Your default inbox token 
if (path.startsWith('/en/')) {
     websiteToken = 'english_token';
} else if (path.startsWith('/es/')) {
     websiteToken = 'spanish_token';
} else if (path.startsWith('/fr/')) {
    websiteToken = 'french_token';
}
// Add more language conditions as needed

(function(d,t) {
    var BASE_URL="https://yourwebsite.com";
    var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
    g.src=BASE_URL+"/packs/js/sdk.js";
    g.defer = true;
    g.async = true;
    s.parentNode.insertBefore(g,s);
    
    g.onload=function(){
        window.chatwootSDK.run({
        websiteToken: websiteToken,
        baseUrl: BASE_URL
        })
    }
})(document,"script");
</script>
```
